src/System.Text.Primitives/System/Text/TextEncoding.cs

```diff
@@ -25,11 +25,13 @@ public enum Id  : byte
 
         public abstract bool TryEncodeFromUtf16(ReadOnlySpan<char> utf16, Span<byte> buffer, out int bytesWritten);
 
-        // TODO: I think we also need to add the following to support transcoding
-        // TryEncodeFromUnicode(ReadOnlySpan<UnicodeCodePoint> codePoints, Span<byte> buffer, out int bytesWritten);
-        // TryEncodeFromUnicode(UnicodeCodePoint codePoint, Span<byte> buffer, out int bytesWritten);
-        // TryDecodeToUnicode(Span<byte> encoded, ReadOnlySpan<UnicodeCodePoint> decoded, out int bytesWritten);
-        // TryDecodeToUnicode(Span<byte> encoded, UnicodeCodePoint decoded, out int bytesWritten);
+        public abstract bool TryEncodeFromUnicode(ReadOnlySpan<UnicodeCodePoint> codePoints, Span<byte> buffer, out int bytesWritten);
+
+        public abstract bool TryEncodeFromUnicode(UnicodeCodePoint codePoint, Span<byte> buffer, out int bytesWritten);
+
+        public abstract bool TryDecodeToUnicode(Span<byte> encoded, Span<UnicodeCodePoint> decoded, out int bytesWritten);
+
+        public abstract bool TryDecodeToUnicode(Span<byte> encoded, out UnicodeCodePoint decoded, out int bytesWritten);
```

Comment:
I know you did not do it in the first place, but could you please split the abstract encoder from the concrete ones (source file wise)?